File: CX.java
import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Scanner;

public final class CX {

    private static String[] args;
    private static Scanner scanner;

    public static void main(String[] args) {
        CX.args = args;
        CX.scanner = new Scanner(System.in);

        String fileIn = argOrInput("file", "请输入要操作的文件(带后缀): ");
        String fileOut = argOrInput("output", "请输入输出的文件名(带后缀): ");

        try {
            byte[] dataIn = Files.readAllBytes(Paths.get(fileIn));
            System.out.println("已读取文件数据");
            byte[] dataOut = cx(dataIn);
            Files.write(Paths.get(fileOut), dataOut);
            System.out.println("已写入数据");
        } catch (FileNotFoundException e) {
            System.out.println("未找到文件: '" + fileIn + "'!");
        } catch (IOException e) {
            System.out.println("在读取时发生了错误!");
        }
    }

    private static byte[] cx(final byte[] bytes) {
        final byte[] result = new byte[bytes.length];
        for (int i = 0; i < bytes.length; i++) {
            if (i % 2 != 0) result[i] = (byte) (bytes[i] ^ 0x64);
            else result[i] = result[i] = (byte) (bytes[i] ^ 0x32);
        }
        return result;
    }

    private static String argOrInput(String argName, String prompt) {
        for (String arg : args) {
            if (arg.startsWith(argName + "=") && arg.length() > argName.length() + 1) {
                return arg.split("=")[1];
            }
        }
        return input(prompt);
    }

    private static String input(String prompt) {
        System.out.print(prompt);
        return scanner.nextLine();
    }

}

File: cx.py
from sys import argv


def arg_or_input(arg_name: str, prompt: str) -> str:
    for item in argv[1:]:
        if item.startswith(arg_name + "="):
            split = item.split("=")
            return input(prompt) if len(split) < 2 else split[1]
    return input(prompt)


def cx(byte: bytes) -> bytes:
    data = b''
    for i, e in enumerate(byte):
        if i % 2 != 0:
            data += bytes([e ^ 0x64])
        else:
            data += bytes([e ^ 0x32])

    return data


f_in = arg_or_input("file", "请输入要操作的文件(带后缀): ")
f_out = arg_or_input("output", "请输入输出的文件名(带后缀): ")

try:
    with open(file=f_in, mode="rb") as f:
        text = f.read()
    print("已读取文件数据")
    with open(file=f_out, mode="wb") as f:
        f.write(cx(text))
    print("已写入数据")

except FileNotFoundError as e:
    print("未找到文件: '" + f_in + "'!")
